_stbt/core.py

```diff
@@ -490,6 +490,19 @@ def __repr__(self):
                 "<Custom Image>" if isinstance(self.image, numpy.ndarray)
                 else repr(self.image)))
 
+    def __nonzero__(self):
+        return self.match
+
+    def __eq__(self, other):
+        return (self.match == other.match and self.region == other.region and
+                self.image == other.image)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __hash__(self):
+        return hash((self.match, self.region, self.image))
```

Comment:
Note that `MatchResult.image` can be a string, or a `numpy.ndarray` (whatever you passed to `stbt.match`). For example you can try to match an image that you saw earlier, instead of matching an image loaded from disk. In that case the `MatchResult` won't be hashable -- whereas before, it would have Python's default hash value, based on the object's id (its address in memory).